.bit/responses/3.2-Week 3 Step 2.md

```diff
@@ -1,80 +1,181 @@
 
-  <br><br/>
 </details>
 
-## **:pencil: Task 2: Call the Giphy API using the translate endpoint.**
+### Writing our *First* Azure Function to Upload an Image
 
-_Note: **Node-fetch** must be installed for the code in this step to work! It should already be installed, but you need to require it in the first lines of code._
+⬇ **Some housekeeping...**
 
-Once you are done, the function should call the Giphy API and **return the data** for a gif that matches the dominant emotion from your emotion data.
+- For the function to work, we have to initialize the packages/libraries we installed in the beginning of part 1.
+- Take note of the `process.env` value being assigned to `connectionstring` in the code below (*Line 3*). Use this [tutorial](https://docs.microsoft.com/en-us/azure/azure-functions/functions-how-to-use-azure-function-app-settings) to add in your own secret strings from your storage container.
+    - The storage container is the one you created in step 1. Navigate to it and find your secret strings.
 
 <details>
-<summary>:exclamation: Can I get a more detailed list of steps?</summary>
-  </br>
-  
-1. In the **async function**, create a new constant that will recieve the dominant emotion from your emotion data (i.e. angry, happy). This dominant emotion should be determined before you call on the trigger function.
-2. Using **node-fetch**, call the Giphy **translate endpoint** from the documentation.
-3. Run the constant through the **string** parameter in the endpoint (look as the documentation if this sounds confusing). Add your **Giphy API key**, and set the number of returned gifs to 1 by adding `&limit=1` to the endpoint.
-4. Declare another constant that formats the results of the API call in **json** format.
-5. Make this second constant the **output** of the function in `context.res`.
-
-  <br><br/>
-  
+<summary>:question: How do I find my secret strings?</summary>
+</br>
+
+![https://user-images.githubusercontent.com/69332964/99161798-ba3d7480-26c3-11eb-8e55-eac4bd4cb174.png](https://user-images.githubusercontent.com/69332964/99161798-ba3d7480-26c3-11eb-8e55-eac4bd4cb174.png)
+
+![https://user-images.githubusercontent.com/69332964/99161822-ec4ed680-26c3-11eb-8977-f12beb496c24.png](https://user-images.githubusercontent.com/69332964/99161822-ec4ed680-26c3-11eb-8977-f12beb496c24.png)
+
+- Keep these safe, and use the connection string in the corresponding variable in the code.
+- *Note: You'll need to store other strings in environment variables as well*
+
 </details>
 
-<details>
-<summary>:exclamation: I'm stuck. Can I get some hints?</summary>
-  </br>
-  
-Fill in the gaps:
-  
+</br>
+
+Let's start just by initializing a few variables we'll need. In your function code at the very top add: 
+
 ```js
+var multipart = require("parse-multipart")
+const connectionString = process.env.AZURE_STORAGE_CONNECTION_STRING;
+const { BlobServiceClient } = require("@azure/storage-blob");
+```
+
+⬇ **The main block of code**

```

Comment:
Split this up into separate drop downs so we don't give students immediately the entire block of code. Maybe into sections like:
- Can we review syntax for parse-multipart?
- How can I determine file extension?
- How can I upload the file?